### DISCUSSION TITLE
OpenApi3 enum values transformed into a scalar and not an enum

### DISCUSSION THREAD

**Discussion by vbossica** [microsoft/typespec#4087] [OpenAPI3]
When defining the OpenApi3 model class:

```
"ModelClass": {
    "enum": [
        "Value1",
        "Value2"
    ],
    "type": "string"
}
```

and using `tsp-openapi3` to transform the specification, the class is generated as:

```
scalar ModelClass extends string;
```

which loses the enum values.

Should the `ModelClass` be generated as an `enum` instead?

```
enum ModelClass {
    Value1,
    Value2
}
```

**Comment by timotheeguerin** [comment]
Yeah that sounds like a bug will convert the discussion
